onnxruntime/test/util/compare_ortvalue.cc

```diff
@@ -232,8 +250,44 @@ std::pair<COMPARE_RESULT, std::string> CompareSeqOfMapToFloat(const T& real_outp
   return std::make_pair(COMPARE_RESULT::SUCCESS, "");
 }
 
-const char* ElementTypeToString(MLDataType type) {
-  return DataTypeImpl::ToString(type);
+std::pair<COMPARE_RESULT, std::string> CompareSparseTensors(const SparseTensor& actual, const SparseTensor& expected,
+                                                            double per_sample_tolerance, double relative_per_sample_tolerance,
+                                                            bool post_processing) {
+  TEST_RETURN_IF_NOT(actual.DataType() == expected.DataType(), COMPARE_RESULT::TYPE_MISMATCH,
+                     "Expected type: ", ElementTypeToString(expected.DataType()),
+                     " actual: ", ElementTypeToString(actual.DataType()));
+
+  TEST_RETURN_IF_NOT(actual.Shape() == expected.Shape(), COMPARE_RESULT::SHAPE_MISMATCH,
+                     "Expected dense shape: ", expected.Shape(),
+                     " Actual: ", actual.Shape());
+
+  TEST_RETURN_IF_NOT(actual.Format() == expected.Format(), COMPARE_RESULT::TYPE_MISMATCH,
+                     "Expected sparse format", expected.Format(),
+                     " actual: ", actual.Format());
+
+  TEST_RETURN_IF_ERROR(CompareTwoTensors(actual.Values(), expected.Values(),
+                                         per_sample_tolerance, relative_per_sample_tolerance, post_processing),
+                       "While comparing sparse values");
```

Comment:
Error message can be improved ?

Reply:
It is a suffix of the error message that only provides context.